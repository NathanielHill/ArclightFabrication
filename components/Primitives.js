import NextLink from 'next/link'
import styled from 'styled-components'

const Center = styled.div`
  align-self: center;
  justify-content: center;
`
const Form = styled.form`
  display: flex;
  flex-direction: column;
`
const H1 = styled.h1`
  font-weight: 300;
`
const H2 = styled.h2`
  font-size: ${props => props.theme.largeFontSize};
`
const Input = styled.input`
  border: 1px solid ${props => props.theme.inputBorder};
  font-family: inherit;
  font-size: ${props => props.theme.baseFontSize};
  margin-bottom: ${props => props.theme.defaultSpacing};
  padding: ${props => props.theme.inputSpacing};
`
const Link = styled(({ children, className, ...rest }) => (<NextLink {...rest}>{children}</NextLink>))`
 cursor: pointer;
`
const P = styled.p`
`
const Select = styled.select`
  background-color: white;
  border: 1px solid ${props => props.theme.inputBorder};
  font-family: inherit;
  font-size: inherit;
  font-size: ${props => props.theme.baseFontSize};
  margin-bottom: ${props => props.theme.defaultSpacing};
  padding: ${props => props.theme.inputSpacing};
`
const TextArea = Select.withComponent('textarea')

export { Center, Form, H1, H2, Input, Link, P, Select, TextArea }
