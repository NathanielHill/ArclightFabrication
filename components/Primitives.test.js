/* global test, expect */

import { Link } from './Primitives'
import React from 'react'
import renderer from 'react-test-renderer'

test('Link renders', () => {
  const component = renderer.create(
    <Link href=''><a>anchor text</a></Link>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
