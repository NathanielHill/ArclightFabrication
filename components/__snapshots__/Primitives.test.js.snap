// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
<a
  href=""
  onClick={[Function]}
>
  anchor text
</a>
`;
